core-blocks/embed/index.js

```diff
@@ -46,6 +40,217 @@ const findBlock = ( url ) => {
 	return 'core/embed';
 };
 
+export function getEmbedEdit( title, icon ) {
+	return class extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.switchBackToURLInput = this.switchBackToURLInput.bind( this );
+			this.setUrl = this.setUrl.bind( this );
+			this.processPreview = this.processPreview.bind( this );
+
+			this.state = {
+				editingURL: false,
+				fetching: !! this.props.attributes.url && ! this.props.preview,
```

Comment:
Can't we get rid of `fetching` using the `isResolving` selector in `core/data`?

Reply:
I don't think so, because (if I'm reading this correctly) `isResolving` works on an attribute in a store, so every embed block on the page would go into fetching mode when one of them fetches.

Reply:
@notnownikki No, it's not, you have to pass the same set of arguments passed to your `getEmbedPreview` selectors to the `isResolving` selector, so you'll pass the `url` of the current block which means you get the "fetching" status for the current selector call.

Reply:
Oh then I had totally read it wrong! I'll get on that :)